### PR TITLE
NODE-1049 only include primary server if there are no secondary serve…

### PR DESCRIPTION
**Pull request by christkv**
…rs for readPrefrence secondaryPreferred

### CODE REVIEW

File: test/tests/functional/rs_mocks/read_preferences_tests.js [hunk 1]
```diff
@@ -1353,3 +1353,221 @@ exports['Should correctly connect connect to single server replicaset and peform
...
+          } else if(doc.count) {
+            request.reply({ "waitedMS" : Long.ZERO, "n" : 1, "ok" : 1});
+          }
+        }
+      }).catch(function(err) {
+        // console.log(err.stack);
+      });
+
+      // Second secondary state machine
+      co(function*() {2
```
Comment (mbroadst) [inline test/tests/functional/rs_mocks/read_preferences_tests.js]: looks like a stray `2` here :)

File: test/tests/functional/rs_mocks/read_preferences_tests.js [hunk 2]
```diff
@@ -1353,3 +1353,221 @@ exports['Should correctly connect connect to single server replicaset and peform
...
+
+                setTimeout(function() {
+                  test.done();
+                }, 500);
+              }
+            });
+          }
+        }, 1000);
+
+        // // Set up a write
```
Comment (mbroadst) [inline test/tests/functional/rs_mocks/read_preferences_tests.js]: is this code meant to be committed?

File: test/tests/functional/rs_mocks/read_preferences_tests.js (hunk 1 above)
Comment (christkv) [inline test/tests/functional/rs_mocks/read_preferences_tests.js]: done

File: test/tests/functional/rs_mocks/read_preferences_tests.js (hunk 2 above)
Comment (christkv) [inline test/tests/functional/rs_mocks/read_preferences_tests.js]: removed

**Pull request merged**